components/Headlines.js

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from "react";
+import { useChannel } from "@ably-labs/react-hooks";
+import styles from "../styles/Home.module.css";
+import Image from "next/image";
+import { formatDate } from "../lib/dates";
+
+/* Subscribes to headline messages from the "news-list" channel
+and provides a form to enter new headlines which it publishes to 
+the same channel */
+export default function Headlines(props) {
+  let inputBox = null;
+  let messageEnd = null;
+
+  const [headlineText, setHeadlineText] = useState("");
+  const [headlines, updateHeadlines] = useState(props.history);
+  const headlineTextIsEmpty = headlineText.trim().length === 0;
+
+  const [channel, ably] = useChannel("headlines", (headline) => {
+    updateHeadlines((prev) => [...prev, headline]);
+  });
+
+  /* Process each message to retrieve the timestamp and author (client Id) */
+  const HeadlinePreviews = ({ items }) => {
```

Comment:
This nested component should probably be split out from the headlines wrapping component. It is surprising to the user seeing it inside another component and makes the file difficult to read as a human.